packages/driver/src/cy/commands/querying.js

```diff
@@ -477,10 +479,42 @@ module.exports = (Commands, Cypress, cy) => {
         options._log.set({ $el })
       }
 
+      // When multiple space characters are considered as a single whitespace in all tags except <pre>.
+      const normalizeWhitespaces = (elem) => {
+        let testText = elem.textContent || elem.innerText || $.text(elem)
+
+        if (elem.tagName === 'PRE') {
```

Comment:
why the exception for `pre` tags? I mean, I can kind of get the logic, but it does seem strange to make an exception for a single tag and this not be documented anywhere. I guess that's my concern. Someone opening an issue one day like, 'this is working differently here' and we have this undocumented exception.

Reply:
`pre` tag preserves whitespaces like this:

https://codepen.io/sainthkh/pen/jOEpzmX

So, it is the exception. Because I thought it reflects how browsers work.

Reply:
Yeah...probably adding a section in the Notes of the `.contains()` doc so that people can find this somehow. There's are other exception notes in there.

Reply:
What about elements styled with `white-space: pre;`?
We have cases where upgrading to Cypress 4 breaks tests where we're explicitily want to vaildate that the whitespace is present as it is rendered. Should there be a flag allowing override of this behaviour?

Reply:
@mariusnn It's going on #6411.